### DISCUSSION TITLE
Fat binary on macOS M1

### DISCUSSION THREAD

**Discussion by tmspzz** [jedisct1/libsodium#1154] [Q&A]
When compiling libsodium from source, is the a way to obtain a fat binary with x86_64 and arm64 archs?

Seems like in some cases JVMs for x84_64 only are asking the linker to get the x84_64 slice on M1 too.

**Comment by jedisct1** [comment]
The `dist-build/apple-xcframework.sh` script builds FAT libraries. You can even directly use the pre-compiled libraries from [swift-sodium](https://github.com/jedisct1/swift-sodium/tree/master/Clibsodium.xcframework).

**Reply by tmspzz** [reply]
Thanks for acknowledging. I will take a look.

On the surface tho if those are .xcframeworks that won't solve the current issue we're facing with .dylib since arch/platforms are still separated and not a FAT binary.

**Reply by peter-toft-greve** [reply]
It would be golden to have Mac support for Universal/fat libs added i.e. removing https://github.com/jedisct1/libsodium/blob/25d2455a42b7c498623aa75b53ea0f16754b4265/configure.ac#L682

**Reply by jedisct1** [reply]
@peter-toft-greve Ignore what was written above. Files in the xcframework are fat libraries.

**Reply by peter-toft-greve** [reply]
@jedisct1 OOh you have my attention - can I generate libsodium using xcframework for MacOS which supports x86_64 and arm64?
Do you have a pointers towards this?

**Reply by jedisct1** [reply]
https://doc.libsodium.org/installation#cross-compiling-to-apple-devices
